src/main/java/org/jabref/logic/importer/fileformat/PdfMergeMetadataImporter.java

```diff
@@ -124,7 +138,9 @@ public ParserResult importDatabase(Path filePath) throws IOException {
                 }
             }
         }
-
+        if (workingDir != null) {
+            filePath = workingDir.relativize(filePath);
+        }
```

Comment:
Call `org.jabref.logic.util.io.FileUtil#relativize` here. 

The required `directories` are provided by `databaseContext.getFileDirectories(filePreferences)`